### PR TITLE
[WIP] Adds Glyphids

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

Adds the glyphids from Deep Rock Galactic (wiki: https://deeprockgalactic.gamepedia.com/Creatures) into SS13 as mobs meant to be controlled by players.  As of current, the mobs themselves have no natural way of spawning and use graphic placeholders, but have all the coding done for them to function in-game if spawned.  The following is what is added:

Glyphid Tower
- A spawner akin to the dens on lavaland which spawns 3 glyphid swarmers.  It will respawn the swarmers slowly when they die or evolve.  Can be destroyed as other spawners can.

Glyphid Swarmer
- Having only 12 health and dealing 2 damage, swarmers are only a threat in very large groups.  They are, however, very fast, and can use that speed to get around very quickly, either to track targets or scout perimeters.  After so much time has passed, they can evolve into one of three castes, a grunt, a silkspinner, or an exploder.  Non-player controlled swarmers will never evolve at the current time, but glyphid swarmers can be possessed similarly to sentient spiders, which then the player can choose to evolve.

Glyphid Grunt:
The essential backbone of the glyphids.  Has the sole ability to build Glyphid Towers, increasing the glyphid numbers.  Creating a tower takes a considerable amount of time (a bit under a minute per tower), so building a massive amount of them is basically impossible alone.  They also deal decent damage and have the highest health of the tier ones, but are slower then the rest of them as well.  They can evolve into Glyphid Veterans.

Glyphid Silkspinner:
The only glyphid whose primary function changed from translating the glyphids to the game.  Instead of just shooting slowing projectiles, the glyphid silkspinner can also lay down biofluid on the floor, which functions identically to a spider's web, but for glyphids.  They have no attack, but instead can heal glyphids by attacking them instead, making them essential for keeping the swarm's higher level members up and running.  They have less health than a grunt, making them a sole support role.

Glyphid Exploder:
A suicidal bomber class which evolves into nothing.  They have 20 health, and unlike the game, they do not explode on death.  This makes them a breaching tool or a high-risk, high-reward unit to take down opposition.  Learn to dodge, or get shot down before you can do anything.  Outside of their singular purpose, the exploder is outclassed in every way by the other glyphids.  They should only be made when the swarm really needs them.

Glyphid Veteran:
A stronger version of the glyphid grunt in combat, having more health and attack damage than the grunt.  Unlike grunts, veterans can hold their ground in combat against more apt opponents.  However, they lack the ability to make Glyphid Towers which make the grunt so useful.  The good thing about this unit is their ability to grow into a Praetorian, which is much more helpful.

Glyphid Acidspitter:
The only main glyphid with a projectile.  Trading in the silkspinner's utility kit in for a solely-ranged one, acidspitters assist the swarm from a range, shooting dangerous projectiles at threats.  Also has more health and has a damaging projectile, unlike the webspitter.  However, the acidspitter's projectile does not slow targets on hit.

Glyphid Praetorian:
The tank of the glyphids.  Packing a thick health amount and considerable damage, these guys allow the lesser glyphid castes to claim an area under their protection.  They also have a much easier time breaking down walls, but they move even slower than the grunt.  If a praetorian lives for a considerable amount of time, they can evolve into a Dreadnought.

Glyphid Dreadnought:
The boss-man of all the glyphids.  Packing an extreme amount of health and 50 damage a bite, dreadnoughts are the ultimate glyphid threat.  They can shoot fireballs, slam the ground to knock nearby foes away, and can break structures as easily as a megafauna.  Working together with other glyphids or alone, the dreadnaught is a massive threat wherever it goes.

Currently, how it works is you'd spawn in a tower somewhere, and players would possess the swarmers and begin filling the ranks.  A drone is a necessity, as towers are both spawn points and area protectors with the glyphid swarmers.  You'd also want a silkspinner to keep the higher tier glyphids from dying, and exploders if the situation calls for it.  If no ghosts are interested in proliferating the glyphids, they'll have little to no impact on the surrounding area, but if enough ghosts fill in the roles, the glyphids can grow to be a sizeable threat.  Using the speed of swarmers to spread out to different areas before setting up as a drone allows Glyphids to expand out and come from different areas, though it might pay off to heavily fortify one area instead.  There is no one way to properly glyphid, allowing for a variety of different strategies using the different castes.

## Why It's Good For The Game

Certainly adds an interesting and varied simplemob into the game which could function as a number of different things in terms of use.  Simple trashmobs for an event or a threat on lavaland, what they could be used for is not exactly set in stone.  They're certainly a varied opponent, however, so however they're introduced will provide entertainment.  They're also a very good ghost role, providing basically an infinite amount of roles to take as long as the towers stay up.  This also ties their standing power into how much people are interested, so if nobody cares about the glyphids, they won't expand far, but greater interest in them will cause them to have a much heavier impact on the round.

However, unlike some simple mobs, getting them to spread isn't automatic, but requires planning and tact in order to so.  Likewise, eliminating the towers and grunts will allow crew to stop the glyphids from proliferating, as no other caste can make the towers.   Another thing to note is the glyphids are not very strong by themselves.  The grunt can be killed simply by being punched to death by a crewmember, let alone the many other robust ways the crew has to eliminate threats.  This design allows a good number of glyphids to be present, as individuals most of the glyphids are very weak.  In the best case scenario, some interesting teamplay will arise to keep a glyphid colony from going under, unlike something like spiders where it just requires a green to lay a few eggs in maint to swarm the entire station.   Finally, their design also gives a goal to strive for and a number of varied ways to let players assist in accomplishing that goal.

## Changelog
:cl:
add: The swarm is upon you!
/:cl:

As noted above, the glyphids do not currently have any natural method of spawning.  I think it's best to see what people would be interested in before committing it to being something specific.  If the interest isn't there, they can still function as an admeme spawn.  Also as noted above, the glyphids and their towers are using placeholder sprites (spider sprites and den sprite) until I see it's worth going forward with them, in which case proper effort can be put into their sprites.  Also, things like their health, damage, and time until possible evolution are not set in stone, and readily available to tweak.

### CODE REVIEW

**Comment by Repukan** [comment]
Should just replace xenos with these

File: code/modules/mob/living/simple_animal/hostile/glyphid.dm [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+	if(.)
+		return
+	humanize_glyphid(user)
+
+/mob/living/simple_animal/hostile/glyphid/proc/humanize_glyphid(mob/user)
+	if(key || !playable_glyphid || stat)//Someone is in it, it's dead, or the fun police are shutting it down
+		return 0
+	var/glyphid_ask = alert("Become a glyphid?", "Do you hate dwarves?", "Yes", "No")
+	if(glyphid_ask == "No" || !src || QDELETED(src))
+		return 1
```
Comment (tralezab) [inline code/modules/mob/living/simple_animal/hostile/glyphid.dm]: use TRUE/FALSE instead of 1/0 for binary returns

**Comment by tralezab** [comment]
i think they are interesting but we have xenos and spiders that fill this role already

**Comment by IndieanaJones** [comment]
Thing is, people hate spiders (as I should know), and xenos can be sometimes problematic.  While the glyphids and the other two all share the 'expand and conquer' style of play, how they accomplish that is all drastically different.  Spiders only require one mother, usually fueled with infinite monkeys from xenobio, and a vent.  Before you realize it, they're everywhere, and there really wasn't anything you could've done to stop it, which usually upsets people.  Kill as many as you'd like, but the one you just killed is just going to go possess another and come right back.  Spiders also have poison, so even if you play very well against them, you will die, and I won't even get into what the special forms can do in the right conditions.  Xenos, on the other hand, are a classic staple, and from that usually get stomped by the station as soon as they show up.  They also have other problems, but I will admit I am no expert.  Glyphids here, and what I believe gives them potential to be better than both, is the different roles the castes fill.  While xenos do this to an extent, the glyphids are much more different from each other than the xenos are.
Each one also plays an important role, and while you can function without some of them, it's better to have all assets on deck.  This goes for both defense and offense as well.  For defense, sure, the drone can build nests, but they don't really have any major way to protect them.  Add in a webspitter, and now you have sticky floors and a healing source to keep your defenders from dying.  For offense, with something like xenos, you'll play footsies with the target before stunning them and dragging them away to get facehugged.  Here, the lone glyphid doesn't have as much power to fuck shit up, but a group of them has many more options.  You could try using acidspitters to get them at a range, or distract them with a praetorian and sending in exploders to flank.  Simply put, there's a lot more diverse encounters to be had when fighting glyphids than the other two.
Also, as a last note, even if the other two fill the "role" of the divide and conquer, as long as each brings it's own unique feel to the table, I'd say they're warranted to exist.  While more diversified content is likely more preferred, nobody is going to keel over if they fight glyphids instead of xenos every once and a while.

File: code/modules/mob/living/simple_animal/hostile/glyphid.dm (hunk 1 above)
Comment (IndieanaJones) [reply code/modules/mob/living/simple_animal/hostile/glyphid.dm]: Resolved.

**Comment by dr-bee** [comment]
Will be fun to test when its ready, and a variety of threats would be nice. This will make a great late round random event.

**Comment by habader** [comment]
Add the swarmers to the gold slime/magicarp pool

**Comment by duckay** [comment]
Maybe this could become some kind of conversion like what Genestealers do on Space Hulks?
Just throwing ideas out there and see what sticks, maybe conversion isn't such a good idea now that I think about it.

**Comment by py01** [comment]
Looks very interesting, though I have a few concerns.
1) Sole support roles tend to be unpopular-people enjoy combat. Usually, standing around building is boring, and it is less fun to click on targets who are willing to stand still for you (healing). I would recommend splitting silkspinner into multiple combat castes each with a single support ability.
2) Wizard Fireball may be a bit too strong. It is a ranged stun that often crits and doesn't need a direct hit. Despite all the other limitations of dreadnought, it might be unfun for the crew.
3) Fighting against the glyphids need to be as fun as being a glyphid, or half the crew will suicide for free antag as soon as they see a glyphid.

**Comment by IndieanaJones** [comment]
For your points:
- Instead of splitting the webspitter's utility into multiple combat classes, what I can do is give the silkspinner a projectile like they have in the game, where they deal low damage but debuff the target.
- The time between fireball shots is pretty long, and the dreadnought will only show up under rare circumstances (the evo. timer is quite long, and you need to progress all the way up from grunt in the first place).  For now, I think them having it will be fine.
- Unlike other mobs, glyphids don't have any stuns nor deal toxin damage.  If you have some protective gear and a weapon, you can go into the fray against them as long as you manage your health, which I believe is more fun than worrying about the xeno stuns or the toxin from spiders.

**Comment by py01** [comment]
Having a channel on the fireball (and a graphical effect to show to others that it is charging) would be a good way to add counterplay to the fireball, since you can bait it out then hide, instead of the current situation where you are likely fucked if you walk near a dreadnought if it happens to have fireball up.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Should just replace xenos with these

wj is literally shaking at this post

**Comment by knacker48** [comment]
I could see this either as its own game mode or maybe a random event like blob or abductors. We'll have to wait and see how balanced they are and what changes have to be made but I'm looking forward to this.

**Comment by IndieanaJones** [comment]
Also, anyone know where to go to nab some spriters?  Is there an area on the forum or Discord I can ask in?

**Comment by DapperMola** [comment]
Would very much like to see these added alongside spiders and xenos or worst case, replace sentient spiders. The only concern I have is the tower, they seem awfully powerful as a spawnpoint for players and defensive simplemobs that's deployed in just under a minute. How strong is 100 integrity relative to other objects?

**Comment by IndieanaJones** [comment]
Not very strong.  These things go down with a few whacks from something like a wrench, for example.

**Comment by IndieanaJones** [comment]
Stuff I added/changed:
- Glyphids are now immune to ash storms
- Webspitter now has a projectile which does no damage but temporarily slows whoever is hit
- Veteran is faster
- Glyphids can ventcrawl, praetorian and dreadnought cannot for obvious reasons.
- Dreadnought now has a delay and warning in chat before he shoots a fireball, in adding this, I found that if he continues to spam click to shoot before the first fireball comes out, more fireballs will result.  While this is technically a bug, it's actually pretty fun so I kept it in.  Considering you now have a warning to get away before the fireballs come out, it seems pretty fair, and allows the Dreadnought to feel more destructive.  Also, if and when the sprites are made, it'll be easy to make an alternative sprite for when he's about to fire.
- Properly toned the exploder's explosion.  Powerful, but only has a range of 2.  Also now catches nearby things on fire.
So, yeah, main focus is finding someone to sprite these bad boys and deciding how they'll show up naturally.  If you guys have any more ideas on that, please do post them.  Also;
> 
> 
> Maybe this could become some kind of conversion like what Genestealers do on Space Hulks?
> Just throwing ideas out there and see what sticks, maybe conversion isn't such a good idea now that I think about it.
I'm unsure what you're talking about, could you clarify?

**Comment by ATH1909** [comment]
If you are gonna make the fireball spam intentional, maybe you should/could add some in-game indication or message that you can do it/that it's not a bug?

**Comment by JMoldy** [comment]
I'd rather replace spiders with this than replace xenos.

Infinite spider eggs are some of the dumbest cancer.

**Comment by ATH1909** [comment]
> I'd rather replace spiders with this than replace xenos.
> 
> Infinite spider eggs are some of the dumbest cancer.

Why does anything have to be replaced?

**Comment by Shadowflame909** [comment]
Because 👏 benos 👏 have 👏 four 👏  ways 👏 to 👏  spawn 👏  and 👏 the 👏 lavaland 👏 ruin 👏 is 👏 the 👏 worst 👏 one.

second worst would be the 8% chance for an egg to spawn in xenobio roundstart.

**Comment by IndieanaJones** [comment]
Well, a guy mentioned in Discord he could do it, so I messaged him and never got anything back.  This might just rot in spriteless hell, who knows.

**Comment by optimumtact** [comment]
this is literally a redo of spiders, which we already have, why would I merge this?

**Comment by Darkgenerallord** [comment]
just replace spiders instead tbh

**Comment by Militaires** [comment]
@optimumtact 

@optimumtact 
**Breaking News: Man Didnt Read Pr**

"I enjoy not reading other's prs are misinterpreting them", said optimumtact, "It really gives me a rush to see people post paragraphs upon paragraphs of text only for me to read the title and formulate a halfassed opinon", local maintainer has stated that they highly enjoy dazzling others unexpectedly by invalidating their efforts, "When I see them post exactly why something is interesting and worthwhile, or explain why two things are different, I quickly take to the github comments to write about how those two things are identical".

**Comment by Militaires** [comment]
also yes spider eggs are bullshit i know because i tc trade for spider midwife role and i 100% ruin the thread ALWAYS without any chance of failure, if even 1 egg cluster hatches that's it the entire station is fucked beyond measure

**Comment by optimumtact** [comment]
so far nobody has attempted any defence of these not just being another playable simple mob that spreads over time.

These are functionally filling the same role as spiders.

**Comment by IndieanaJones** [comment]
I can copy and paste the response from up above again if you'd like:
Thing is, people hate spiders (as I should know), and xenos can be sometimes problematic. While the glyphids and the other two all share the 'expand and conquer' style of play, how they accomplish that is all drastically different. Spiders only require one mother, usually fueled with infinite monkeys from xenobio, and a vent. Before you realize it, they're everywhere, and there really wasn't anything you could've done to stop it, which usually upsets people. Kill as many as you'd like, but the one you just killed is just going to go possess another and come right back. Spiders also have poison, so even if you play very well against them, you will die, and I won't even get into what the special forms can do in the right conditions. Xenos, on the other hand, are a classic staple, and from that usually get stomped by the station as soon as they show up. They also have other problems, but I will admit I am no expert. Glyphids here, and what I believe gives them potential to be better than both, is the different roles the castes fill. While xenos do this to an extent, the glyphids are much more different from each other than the xenos are.
Each one also plays an important role, and while you can function without some of them, it's better to have all assets on deck. This goes for both defense and offense as well. For defense, sure, the drone can build nests, but they don't really have any major way to protect them. Add in a webspitter, and now you have sticky floors and a healing source to keep your defenders from dying. For offense, with something like xenos, you'll play footsies with the target before stunning them and dragging them away to get facehugged. Here, the lone glyphid doesn't have as much power to fuck shit up, but a group of them has many more options. You could try using acidspitters to get them at a range, or distract them with a praetorian and sending in exploders to flank. Simply put, there's a lot more diverse encounters to be had when fighting glyphids than the other two.
Also, as a last note, even if the other two fill the "role" of the divide and conquer, as long as each brings it's own unique feel to the table, I'd say they're warranted to exist. While more diversified content is likely more preferred, nobody is going to keel over if they fight glyphids instead of xenos every once and a while.

**Comment by FantasticFwoosh** [comment]
I apologise for not reading the above paragraph, but i welcome glyphids. Spiders/xeno's are generally pretty mundane and spiders are only seen via xenobio and literally defeated by the AI venting out oxygen, adding these would allow us to do more with them like a dedicated 'infestation' mode to suppliment xenos/ spider events that feels genuinely alien and who knows where the future lies.

- They might have a better life on TGCM if they can be converted to carbons, or creative uses are sought out for them, like having a genestealer esque humanoid carbon high level evolution to function at the level of a ashwalker, roughly.

- Its also less annoying but relatively equal to the removed swarmer lavaland ruin if we can get some autonomous AI on them and incentivise killing towers/mounds for loot.

Oranges doesn't like long paragraphs and is known for not reading it all so you might have to paraphrase your points to get them across in more digestable chunks OP.

**Comment by DapperMola** [comment]
>so far nobody has attempted any defence of these not just being another playable simple mob that spreads over time.
>
>These are functionally filling the same role as spiders.

if you read just about anything in this PR, you would know thats not true.

**Comment by optimumtact** [comment]
You can repeat the same arguments but that won't make me change my mind any more than it did the first time.

I still see no reason why this shouldn't replace spiders.

**Comment by IndieanaJones** [comment]
Well, you would be removing the variety that comes with adding new content if you just remove stuff to compensate, for starters.  Another argument would be if they can replace spiders, why not xenos?  Honestly, I think the way I'd have them spawn is to replace the xeno spawn on Lavaland with these guys instead, or something along those lines.  I kinda think xenos don't mesh with the rest of Lavaland, while these guys seem to belong more.  If we go down that route, we can also add the ability to mine to the Glyphids so they can build tunnels and such as well.  This would also give them a purpose to leave be as they would mine out things as they expand, which could then be taken later.

**Comment by optimumtact** [comment]
unrelated but would you please learn to use paragraphs

**Comment by IndieanaJones** [comment]
There you are <3

**Comment by IndieanaJones** [comment]
Until this is prepared with proper sprites, I'll be closing this and we can pick up the discussion then.  Thank you to everyone who voiced their opinion.

**Comment by ExcessiveUseOfCobblestone** [comment]
l0l

**Pull request closed without merging**